### ISSUE TITLE
__repr__ fails for jax.ShapeDtypeStruct returned by eval_shape

### ISSUE THREAD

**Issue by Conchylicultor** [google/jax#2320]
Example to reproduce the issue:

```python
import jax
import jax.numpy as jnp

out_spec = jax.eval_shape(lambda x: x, jax.ShapeDtypeStruct(shape=(3,), dtype=jnp.float32))
print(out_spec)  # raise AttributeError: 'numpy.dtype' object has no attribute 'dtype'>
```

`out_spec.dtype` contains the onp.dtype, while `__repr__` try to print `self.dtype.dtype.name`

**Comment by gnecula** [comment]
Thanks for reporting this, we have a fix in progress #2253

**Comment by gnecula** [comment]
Duplicate #2253
